### PR TITLE
ADD basic bitmap:

### PR DESCRIPTION
**Pull request by riblanc**
Adding a basic bitmap using an array of usize to store bits information

### CODE REVIEW

File: src/memory/bitmap.zig [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+
+		self.* = BitMap{};
+		self.nb_obj = nb_obj;
+		self.bits = addr[0..len];
+		for (self.bits) |*b| b.* = 0;
+	}
+
+	// return the number of usize needed to store nb_obj bits
+	pub fn compute_len(nb_obj: usize) usize {
+		return ft.mem.alignForward(usize, nb_obj, nbits) / nbits;
```
Comment (jgiron42) [inline src/memory/bitmap.zig:28]:
```suggestion
		return ft.math.divCeil(usize, nb_obj, nbits) catch unreachable;
```

File: src/memory/bitmap.zig [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+	pub const Error = error{ OutOfBounds };
+
+	pub fn init(self: *Self, addr: [*]usize, nb_obj: usize) void {
+		var len = Self.compute_len(nb_obj);
+
+		self.* = BitMap{};
+		self.nb_obj = nb_obj;
+		self.bits = addr[0..len];
+		for (self.bits) |*b| b.* = 0;
+	}
```
Comment (jgiron42) [inline src/memory/bitmap.zig:24]: I think the "norm" in zig for the init function is to be a "static" function that return the object like in https://ziglang.org/documentation/master/std/src/std/Random.zig.html#L35

File: src/memory/bitmap.zig [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+
+	pub const Error = error{ OutOfBounds };
+
+	pub fn init(self: *Self, addr: [*]usize, nb_obj: usize) void {
+		var len = Self.compute_len(nb_obj);
+
+		self.* = BitMap{};
+		self.nb_obj = nb_obj;
+		self.bits = addr[0..len];
+		for (self.bits) |*b| b.* = 0;
```
Comment (jgiron42) [inline src/memory/bitmap.zig:23]: why not memset?

File: src/memory/bitmap.zig [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+	}
+
+	// return the number of usize needed to store nb_obj bits
```
Comment (jgiron42) [inline src/memory/bitmap.zig:26]: these comments should be documentation comments
```suggestion
	/// return the number of usize needed to store nb_obj bits
```

**Review by jgiron42 — CHANGES_REQUESTED** [review]

File: src/memory/bitmap.zig (hunk 3 above)
Comment (riblanc) [reply src/memory/bitmap.zig:23]: Honestly, don't know :shrug:

File: src/memory/bitmap.zig (hunk 2 above)
Comment (riblanc) [reply src/memory/bitmap.zig]: Done 8a8dcb3

File: src/memory/bitmap.zig (hunk 3 above)
Comment (riblanc) [reply src/memory/bitmap.zig]: Done 8a8dcb3

File: src/memory/bitmap.zig (hunk 4 above)
Comment (riblanc) [reply src/memory/bitmap.zig]: Done 8a8dcb3

**Review by jgiron42 — APPROVED** [review]

**Pull request merged**